src/Application/Common/DbSetExtensions.cs

```diff
@@ -0,0 +1,17 @@
+using System.Linq.Expressions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.EntityFrameworkCore.Query;
+
+namespace Crpg.Application.Common;
+
+public static class DbSetExtensions
+{
+    public static async Task<int> RemoveRangeAsync<T>(this DbSet<T> entitySet,  Expression<Func<T, bool>> predicate, CancellationToken cancellationToken = default) where T : class
+    {
+        var list = await entitySet.Where(predicate).ToArrayAsync();
+        entitySet.RemoveRange(list);
+
+        return list.Count();
+    }
+}
```

Comment:
This is not Needed. In entity Framework Core 7.0.0 you can use
Where(ei => ei.UserItem!.ItemId == req.ItemId)
                .ExecuteDeleteAsync();

![image](https://github.com/namidaka/crpg/assets/47090987/e67ae44a-a186-4ea5-bd0a-23679fdc89cb)

Reply:
ExecuteDelete* is not supported for the In-Memory provider, we should switch to sqlite? provider for tests.

Reply:
> ExecuteDelete* is not supported for the In-Memory provider, we should switch to sqlite? provider for tests.

You're right. I tried migrating to 8.0.0 see if it was working but no.
Sqlite does not support storing objects , so it fails all the test.
You can ignore my previous comment for the moment.